editoast/src/models/work_schedules.rs

```diff
@@ -22,6 +32,92 @@ pub struct WorkScheduleGroup {
     pub name: String,
 }
 
+// pub trait ModelColumn<M: super::Model> {
+//     type ModelType;
+
+//     fn eq(&self, value: Self::ModelType) -> FilterSetting<M>;
+// }
+
+// impl WorkScheduleGroup {
+//     pub const fn name() -> impl ModelColumn<Self, ModelType = String> {
+//         struct Name;
+//         impl ModelColumn<WorkScheduleGroup> for Name {
+//             type ModelType = String;
+
+//             fn eq(&self, value: Self::ModelType) -> FilterSetting<WorkScheduleGroup> {
+//                 use diesel::ExpressionMethods as _;
+//                 FilterSetting::new(editoast_models::tables::work_schedule_group::name.eq(value))
+//             }
+//         }
+//         Name
+//     }
+// }
+
+// pub trait ModelColumn2<M: super::Model, Column, ModelType>
+// where
+//     Column: diesel::ExpressionMethods
+//         + diesel::query_builder::QueryFragment<diesel::pg::Pg>
+//         + ValidGrouping<()>
+//         + Send
+//         + 'static,
+//     Column::SqlType: SqlType,
+//     <Column::SqlType as SqlType>::IsNull: OneIsNullable<<Column::SqlType as SqlType>::IsNull>,
+//     <Column as ValidGrouping<()>>::IsAggregate:
+//         MixedAggregates<diesel::expression::is_aggregate::No>,
+//     ModelType: diesel::expression::AsExpression<Column::SqlType>,
+//     ModelType::Expression: diesel::query_builder::QueryFragment<diesel::pg::Pg> + Send + 'static,
+// {
+//     const COLUMN: Column;
+
+//     // fn eq_(&self, other: Self::ModelType) -> diesel::dsl::Eq<Column, Self::ModelType> {
+//     //     Self::COLUMN.eq(other)
+//     // }
+
+//     fn eq(&self, other: ModelType) -> FilterSetting<M>
+//     where
+//         <<<Column as diesel::Expression>::SqlType as SqlType>::IsNull as OneIsNullable<
+//             <<Column as diesel::Expression>::SqlType as SqlType>::IsNull,
+//         >>::Out: diesel::sql_types::MaybeNullableType<diesel::sql_types::Bool>,
+//     {
+//         let f = Self::COLUMN.eq(other);
+//         FilterSetting::new(f)
+//     }
+
+//     fn asc(&self) -> SortSetting<M>
+//     where
+//         <Column as ValidGrouping<()>>::IsAggregate:
+//             MixedAggregates<diesel::expression::is_aggregate::No>,
+//         Column: SelectableExpression,
+//     {
+//         let s = Self::COLUMN.asc();
+//         SortSetting(Box::new(s))
+//     }
+// }
+
+// impl WorkScheduleGroup {
+//     pub const fn name2(
+//     ) -> impl ModelColumn2<Self, editoast_models::tables::work_schedule_group::name, String> {
+//         struct Name;
+//         impl
+//             ModelColumn2<
+//                 WorkScheduleGroup,
+//                 editoast_models::tables::work_schedule_group::name,
+//                 String,
+//             > for Name
+//         {
+//             const COLUMN: editoast_models::tables::work_schedule_group::name =
+//                 editoast_models::tables::work_schedule_group::name;
+//         }
+//         Name
+//     }
+// }
+
+// fn lolz() {
```

Comment:
Nice bit of code there

Reply:
Isn't it? Peak programming practices around here.

Reply:
(Thanks, I thought I staged this 🐒)